cmd/manager/main.go

```diff
@@ -125,11 +134,39 @@ func main() {
 		os.Exit(1)
 	}
 
+	hubNamespace := os.Getenv("NAMESPACE")
+	if len(hubNamespace) == 0 {
+		inClusterNamespace, err := util.GetInClusterNamespace()
+		if err != nil {
+			setupLog.Error(err, "the manager should be either running in a container or specify NAMESPACE environment")
+		}
+		hubNamespace = inClusterNamespace
+	}
+
+	if len(imagePullSecretName) == 0 {
+		imagePullSecretName = os.Getenv("AGENT_IMAGE_PULL_SECRET")
+	}
+	imagePullSecret := &corev1.Secret{}
+	if len(imagePullSecretName) != 0 {
+		imagePullSecret, err = nativeClient.CoreV1().Secrets(hubNamespace).Get(
+			context.TODO(),
+			imagePullSecretName,
+			metav1.GetOptions{},
+		)
+		if err != nil {
+			setupLog.Error(err, "fail to get agent image pull secret")
+		}
+		if imagePullSecret.Type != corev1.SecretTypeDockerConfigJson {
+			setupLog.Error(errors.Errorf("incorrect type for agent image pull secret"), "")
+		}
```

Comment:
should we exit here? Or we ignore the error and still start a controller with nil secret?

Reply:
good question... i think the behavior of other similar addon controller is to exit here 

@qiujian16 would it be acceptable?

Reply:
updated and added `os.Exit(1)` for when secret does not exist and when secret is not of the right type